subsys/nrf_security/src/drivers/cracen/cracenpsa/src/ecdsa.c

```diff
@@ -0,0 +1,554 @@
+			sx_pk_release_req(pkreq.req);
+			if (i == MAX_ECDSA_ATTEMPTS) {
+				return SX_ERR_TOO_MANY_ATTEMPTS;
+			}
+		} else {
+			break;
+		}
+	}
+	if (status != SX_OK) {
+		sx_pk_release_req(pkreq.req);
+		return status;
+	}
+
+	const char **outputs = sx_pk_get_output_ops(pkreq.req);
+
+	ecdsa_read_sig(&internal_signature, outputs[0], outputs[1], opsz);
+	sx_pk_release_req(pkreq.req);
+	safe_memzero(workmem, workmem_requirement);
+
+	return SX_OK;
+}
+
+static int run_deterministic_ecdsa_hmac_step(const struct sxhashalg *hashalg, size_t opsz,
+					     const char *curve_n, size_t digestsz, size_t blocksz,
+					     char *workmem, struct ecdsa_hmac_operation *hmac_op,
+					     const struct ecc_priv_key *privkey);
+
+static int deterministic_ecdsa_hmac(const struct sxhashalg *hashalg, const uint8_t *key,
+				    const uint8_t *v, size_t hash_len,
+				    internal_octet_t internal_octet, const uint8_t *sk,
+				    const uint8_t *hash, size_t key_len, uint8_t *hmac);
+
+int cracen_ecdsa_sign_digest_deterministic(const struct ecc_priv_key *privkey,
+					   const struct sxhashalg *hashalg,
+					   const struct sx_pk_ecurve *curve, const uint8_t *digest,
+					   size_t digestsz, uint8_t *signature)
+{
+	int status;
+	int attempts = MAX_ECDSA_ATTEMPTS;
+	size_t opsz = (size_t)sx_pk_curve_opsize(curve);
+	size_t blocksz = sx_hash_get_alg_blocksz(hashalg);
+	size_t workmem_requirement = digestsz * 4 + opsz + blocksz;
+	const char *curve_n = sx_pk_curve_order(curve);
+	char workmem[workmem_requirement];
+
+	struct sx_pk_acq_req pkreq;
+	struct sx_pk_inops_ecdsa_generate inputs;
+	struct ecdsa_signature internal_signature = {0};
+
+	internal_signature.r = signature;
+	internal_signature.s = signature + opsz;
+	memcpy(workmem, digest, digestsz);
+
+	do {
+		struct ecdsa_hmac_operation hmac_op = {0};
+
+		hmac_op.attempts = MAX_ECDSA_ATTEMPTS;
+
+		uint8_t *h1 = workmem + digestsz + blocksz;
+
+		memcpy(h1, workmem, digestsz);
+		while (hmac_op.step <= DETERMINISTIC_HMAC_STEPS) {
+			status = run_deterministic_ecdsa_hmac_step(hashalg, opsz, curve_n, digestsz,
+								   blocksz, workmem, &hmac_op,
+								   privkey);
+			if (status != SX_OK && status != SX_ERR_HW_PROCESSING) {
+				return status;
+			}
+		}
+
+		pkreq = sx_pk_acquire_req(SX_PK_CMD_ECDSA_GEN);
+		if (pkreq.status) {
+			return pkreq.status;
+		}
+		pkreq.status =
+			sx_pk_list_ecc_inslots(pkreq.req, curve, 0, (struct sx_pk_slot *)&inputs);
+		if (pkreq.status) {
+			return pkreq.status;
+		}
+		ecdsa_write_sk(privkey, inputs.d.addr, opsz);
+		sx_wrpkmem(inputs.k.addr, workmem + digestsz, opsz);
+		digest2op(workmem, digestsz, inputs.h.addr, opsz);
+		sx_pk_run(pkreq.req);
+		status = sx_pk_wait(pkreq.req);
+		if (status != SX_OK) {
+			sx_pk_release_req(pkreq.req);
+		}
+
+	} while ((attempts--) && ((pkreq.status == SX_ERR_INVALID_SIGNATURE) ||
+				  (pkreq.status == SX_ERR_NOT_INVERTIBLE)));
+	const char **outputs = sx_pk_get_output_ops(pkreq.req);
```

Comment:
Does it make sense to add a check that the status was SX_OK here and that the pkreq.status is also OK?
It seems to me that we ignore the error codes here, no?

Reply:
I believe pkreq.status will always be OK in this scenario, because after sx_pk_wait the operation is already finished. All this operation is doing is retrieving stored data from a successful operation.  I can however add it in if it is still wanted, I based this on how previously existing code has handled the operation, but that might very well be wrong

Reply:
It just reads wrong to override the error code with just SX_OK to me. NOt exactly a blocker but it seems to reasonable to me to have the check.

Reply:
I checked the code. pkreq.status is only set by `ba414ep_is_busy` and `sx_pk_get_status` which is either called directly or by `sx_pk_acquire_req`. So I don't think it makes sense to check `pkreq.status` here as it won't be changed. 
`ecdsa_read_sig(&internal_signature, outputs[0], outputs[1], opsz);`
`sx_pk_release_req(pkreq.req);`
`safe_memzero(workmem, workmem_requirement);`
are all void so we don't drop any status from that and we always check status before this, so I don't see where we are dropping the status? Basically `sx_pk_wait(pkreq.req)` will be return an error code or this function is OK, unless I have missed something?

Reply:
Also only a bit related, sometimes we set the status ourselves with: `pkreq.status = sx_pk_list_ecc_inslots(pkreq.req, curve, 0, (struct sx_pk_slot *)&inputs)` or some other function, but those cases we also return immediately

Reply:
Discussed offline, this will get updated to avoid ignoring the status code.